src/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -747,7 +747,8 @@ Element createParameterElement( final ParameterDefinitionEntry parameter,
 
             parameterElement.appendChild( attributeElement );
 
-            if ( ParameterAttributeNames.Core.POST_PROCESSOR_FORMULA.equals( attributeName )
+            if ( ( ( attributeValue.equals( "textbox" ) || attributeValue.equals( "multi-line" ) ) && !Number.class.isAssignableFrom( valueType ) && valueType != String.class )
```

Comment:
Dates can be validated on the client too. See my comment on the common-ui pull request for that.

Once a condition spans multiple lines, it is a good time to extract it into a new method: 

protected bool isParameterServerValidated(string attrName, string attrValue) {
  ..
} 

in there you can make the many checks a lot more readable than trying to cram everything into a single statement. 

Also: There are constants defined for the various parameter-presentation types. (ParameterAttributeNames.Core.TYPE_TEXTBOX, etc), please use them so that future maintenance work never has a chance to introduce typos.